### PR TITLE
refactor: use single entry point for reveal

### PR DESCRIPTION
**Pull request by christjt**
#### Type of change

Refactor 

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/REV-485

## Description :pencil:

Moves /tools and /extensions to index.ts such that we only have a single entry point. This allows for smaller bundle size as well as enabling tools such as API Extractor, going forward. 

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: documentation/docs/api_readme.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
 `@cognite/reveal` has three modules:
 - [@cognite/reveal](#module-cognitereveal) is the main module and has the main entry point of Reveal, [`Cognite3DViewer`](#class-cognite3dviewer). CAD models are represented by [`CogniteCadModel`](#class-cognitecadmodel) and points clouds by [`CognitePointCloudModel`](#class-cognitepointcloudmodel).
 - [@congite/reveal/tools](#module-cogniterevealtools) contains a set of tools, e.g. [`AxisViewTool`](#class-axisviewtool), [`GeomapTool`](#class-geomaptool), [`TimelineTool`](#class-timelinetool), [`ExplodedViewTool`](#class-explodedviewtool) and [`HtmlOverlayTool`](#class-htmloverlaytool) that works with `Cognite3DViewer`.
-- [@cognite/reveal/extensions/datasource](#module-cogniterevealextensionsdatasource) allows providing custom data sources for geometry and model metadata.
```
Comment (haakonflatval-cognite) [inline documentation/docs/api_readme.md:4]: I guess we'll have to update this file now :sweat_smile:

File: examples/src/utils/MeasurementUi.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
+import { THREE } from '@cognite/reveal';
+import dat from "dat.gui";
 import { Cognite3DViewer } from "@cognite/reveal";
 import { MeasurementTool } from "@cognite/reveal/tools";
-import dat from "dat.gui";
-import { THREE } from '@cognite/reveal';
 
 export class MeasurementUi {
   private readonly _viewer: Cognite3DViewer;
```
Comment (haakonflatval-cognite) [inline examples/src/utils/MeasurementUi.ts:7]: Just curious - do we have a convention on the order of imports in places like these?

File: viewer/jest.config.js [hunk 3]
```diff
@@ -22,9 +22,6 @@ module.exports = () => {
       '\\.svg$': path.resolve(__dirname, './test-utilities/src/filetype-mocks/svgMock.js'),
       '\\.wasm$': path.resolve(__dirname, './test-utilities/src/filetype-mocks/wasmMock.js')
     },
-    globals: {
-      __webpack_public_path__: ''
-    },
```
Comment (haakonflatval-cognite) [inline viewer/jest.config.js:27]: Now I'm really starting to wonder what this is :monocle_face:

**Review by haakonflatval-cognite — APPROVED** [review]
Looks nice - a couple of comments/questions

File: examples/src/utils/MeasurementUi.ts (hunk 2 above)
Comment (christjt) [reply examples/src/utils/MeasurementUi.ts:7]: Just feel like external import should be first. No reason really, we should probably just add some lint rule for this.

File: viewer/jest.config.js (hunk 3 above)
Comment (christjt) [reply viewer/jest.config.js:27]: Path to CDN for V8 workers.

File: documentation/docs/api_readme.md (hunk 1 above)
Comment (christjt) [reply documentation/docs/api_readme.md:4]: Yeah, I messed up here, I see. Need to document these changes.

**Pull request merged**